internal/util/relations.go

```diff
@@ -1,11 +1,29 @@
 package util
 
+import "sigs.k8s.io/controller-runtime/pkg/client"
+
+// FR represents a foreign relation.
+type FR struct {
+	Identifier string
+	// Referer represents K8s object that references a Plugin.
+	Referer client.Object
+}
```

Comment:
I'm OK leaving this in in this PR but IMHO we should move all this code out to a dedicated package at some point where all foreign relations code is put. WDYT?

Reply:
This is a sensible idea to do in a separate PR